From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Manish Bhardwaj <mbhardwaj@nvidia.com>
Date: Wed, 14 Jan 2026 04:55:59 +0000
Subject: [PATCH 131/241] NVIDIA: SAUCE: pinctrl: tegra: add tegra_hv.h header
 file

Previously, tegra_hv.c driver was having hypervisor
syscall interface, hypervisor IVC mempool and hypervisor
IVC queue implementation.

Now refactor the tegra_hv.c driver into tegra_hv.c driver
and tegra_hv_ivc.c driver where tegra_hv_ivc.c driver will
have hypervisor IVC queue impelemtation and reset in tegra_hv.c
so update the header files based on refactoring

Jira ESLC-8959

Change-Id: I1431fc938c4beea54f719747152ba3ca9313383f
Signed-off-by: Manish Bhardwaj <mbhardwaj@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3532104
Reviewed-by: svcacv <svcacv@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Brad Griffis <bgriffis@nvidia.com>
(cherry picked from commit 2fa4b2850fb6d5a3441f728746a1be1c13689555)

diff --git a/drivers/pinctrl/tegra/pinctrl-tegra.c b/drivers/pinctrl/tegra/pinctrl-tegra.c
index 2282a96af025..8eacfe589897 100644
--- a/drivers/pinctrl/tegra/pinctrl-tegra.c
+++ b/drivers/pinctrl/tegra/pinctrl-tegra.c
@@ -1,4 +1,5 @@
 // SPDX-License-Identifier: GPL-2.0-only
+// SPDX-FileCopyrightText: Copyright (c) 2011-2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 /*
  * Driver for the NVIDIA Tegra pinmux
  *
@@ -23,7 +24,10 @@
 #include <linux/pinctrl/pinconf.h>
 #include <linux/pinctrl/pinctrl.h>
 #include <linux/pinctrl/pinmux.h>
-#include <soc/tegra/virt/hv-ivc.h>
+
+#if IS_ENABLED(CONFIG_TEGRA_HV_DRIVER)
+#include <soc/tegra/virt/tegra_hv.h>
+#endif
 
 #include "../core.h"
 #include "../pinctrl-utils.h"
-- 
2.43.0

